vadin-demo-gui/src/main/java/de/muenchen/vaadin/ui/components/BuergerGrid.java

```diff
@@ -1,69 +1,221 @@
 package de.muenchen.vaadin.ui.components;
 
-import com.vaadin.data.Item;
-import com.vaadin.data.util.BeanItemContainer;
-import com.vaadin.data.util.GeneratedPropertyContainer;
-import com.vaadin.data.util.PropertyValueGenerator;
-import com.vaadin.ui.CustomComponent;
-import com.vaadin.ui.Grid;
+import com.vaadin.data.util.BeanItem;
+import com.vaadin.event.ShortcutAction;
+import com.vaadin.server.FontAwesome;
+import com.vaadin.ui.*;
+import com.vaadin.ui.themes.ValoTheme;
+import de.muenchen.eventbus.events.AppEvent;
+import de.muenchen.eventbus.types.EventType;
 import de.muenchen.vaadin.demo.api.domain.Buerger;
-import de.muenchen.vaadin.guilib.util.VaadinUtil;
+import de.muenchen.vaadin.demo.api.local.LocalBuerger;
+import de.muenchen.vaadin.demo.i18nservice.buttons.ActionButton;
+import de.muenchen.vaadin.demo.i18nservice.buttons.SimpleAction;
+import de.muenchen.vaadin.guilib.components.GenericGrid;
+import de.muenchen.vaadin.ui.app.views.BuergerCreateView;
+import de.muenchen.vaadin.ui.app.views.BuergerDetailView;
+import de.muenchen.vaadin.ui.app.views.BuergerUpdateView;
 import de.muenchen.vaadin.ui.controller.BuergerViewController;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import static de.muenchen.vaadin.demo.i18nservice.I18nPaths.Type;
 import static de.muenchen.vaadin.demo.i18nservice.I18nPaths.getEntityFieldPath;
 
 /**
  *
- * @author claus.straube
+ * @author claus.straube, arne.schoentag
  */
 public class BuergerGrid extends CustomComponent {
 
-    private final BeanItemContainer<Buerger> container;
+    protected static final Logger LOG = LoggerFactory.getLogger(BuergerGrid.class);
     private BuergerViewController controller;
-    private Grid grid;
+
+    private GenericGrid grid;
+    private TextField filter = new TextField();
+    private Button search;
+    private Button reset;
+    private ActionButton edit;
+    private ActionButton copy;
+    private ActionButton delete;
 
     public BuergerGrid(final BuergerViewController controller) {
```

Comment:
Das ist m.M. nach sehr komplex und könnte in mehrere private Methoden aufgeteilt werden